### PR TITLE
Ensure valid application pointer in RAI::CheckLanguage and FinishSendingResponseToMobile

### LINKED ISSUE #3813: Core assumes application will remain valid thru registration process
### Bug Report
It was noticed that Core assumes `applications_` entry will be valid throughout the registration process, when the `applications_` entry may be removed.

##### Reproduction Steps
1. Register some apps
2. Change language quickly

##### Expected Behavior
Registration process completes successfully

##### Observed Behavior
`application == nullptr` in either of the following methods: (depending on timing)

```
commands::RegisterAppInterfaceRequest::CheckLanguage()
sdl_rpc_plugin::commands::FinishSendingResponseToMobile(...)
```

##### OS & Version Information
* OS/Version: ubuntu 18.04
* SDL Core Version: develop
* Testing Against: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2602

### PR DESCRIPTION
**Pull request by iCollin**
Fixes #3813

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2602

### Summary
Prevent derefing nullptr for application in some methods during registration

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Review by jacobkeeler — COMMENTED** [review]
Should this check be added to `ApplicationDataShouldBeResumed` as well?

**Comment by iCollin** [comment]
Yes good theory, we should also probably check if application has been unregistered before beginning `ResumptionDataProcessorImpl::Restore`.

File: src/components/application_manager/src/resumption/resume_ctrl_impl.cc [hunk 1]
```diff
@@ -558,7 +558,6 @@ void ResumeCtrlImpl::ResetLaunchTime() {
 bool ResumeCtrlImpl::CheckPersistenceFilesForResumption(
     ApplicationSharedPtr application) {
   SDL_LOG_AUTO_TRACE();
-  DCHECK_OR_RETURN(application, false);
```
Comment (jacobkeeler) [inline src/components/application_manager/src/resumption/resume_ctrl_impl.cc:561]: Not certain whether we need to remove these checks, I realize that they might be slightly overkill, but realistically they don't hurt to have

**Review by jacobkeeler — CHANGES_REQUESTED** [review]

File: src/components/application_manager/src/resumption/resume_ctrl_impl.cc (hunk 1 above)
Comment (iCollin) [reply src/components/application_manager/src/resumption/resume_ctrl_impl.cc]: restored in https://github.com/smartdevicelink/sdl_core/pull/3815/commits/bed8d7af1592023f3ce8578754c5d96b0b8bbe4a

**Review by jacobkeeler — APPROVED** [review]

**Pull request merged**